examples/full_tutorial.ipynb

```diff
@@ -0,0 +1,882 @@
+    "        </button>\n",
+    "    };\"\"\"\n",
+    "m = MyCounterWidget()\n",
+    "m"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5d45c672",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m.value # <- like this you can access the value from python"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1827bb2b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m.value = 20 # <- setting the value also works here."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "67827f3d",
+   "metadata": {},
+   "source": [
+    "next, let's make a **simple traitlet oberservation using `change`** "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b23f3e06",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from traitlets import  Any, observe\n",
+    "from traitlets import Int, Any\n",
+    "import ipyreact\n",
+    "\n",
+    "class Example(ipyreact.ReactWidget):\n",
+    "    count = Int(0).tag(sync=True)\n",
+    "\n",
+    "    @observe(\"count\")\n",
+    "    def _observe_count(self, change):\n",
+    "        print(f\"Old value: {change['old']}\")\n",
+    "        print(f\"New value: {change['new']}\")\n",
+    "        print(f\"--------------------------\")\n",
+    "\n",
+    "    _esm = \"\"\"\n",
+    "    import * as React from \"react\";\n",
+    "\n",
+    "    export default function({on_count, count, prime_message}) {\n",
+    "        return <div><button onClick={() => on_count(count + 1)}>\n",
+    "            {count} times clicked \n",
+    "        </button>\n",
+    "        </div>\n",
+    "    };\"\"\"\n",
+    "\n",
+    "\n",
+    "Example()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "41e3e447",
+   "metadata": {},
+   "source": [
+    "next, let's **observe a traitlet and call python function** everytime the value of that traitlet changes.  \n",
+    "First we need a python function, e.g. in our example it will tell us if we have a prime number or not:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e1b5af36",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from traitlets import  Any, observe\n",
+    "from traitlets import Int, Any\n",
+    "import ipyreact\n",
+    "\n",
+    "def is_prime_number(n):\n",
+    "    for i in range(2, n):\n",
+    "        if n % i == 0:\n",
+    "            return \"No 💻🧊🧊🧊\"\n",
+    "    return \"Yes 💻✅✅✅\"\n",
+    "\n",
+    "\n",
+    "class PrimePythonWidget(ipyreact.ReactWidget):\n",
+    "    prime_message = Any(\"Click the Button\").tag(sync=True)\n",
+    "    count = Int(0).tag(sync=True)\n",
+    "\n",
+    "    @observe(\"count\")\n",
+    "    def _observe_count(self, change):\n",
+    "        self.prime_message = is_prime_number(self.count) # <-  TODO find out if self.count is same as change[\"new\"]\n",
```

Comment:
In my experience they are the same.  I only use `change["new"]` when I also need to reference `change["old"]`

Reply:
indeed, i prefer to explicitly use self.count and ignore the change argument. This change argument is only useful if a function is bound as an observed that us an unbound method, which does not have access to 'self', and it useful across multiple objects/classes.